tests/test_group_target_encoder.py

```diff
@@ -0,0 +1,215 @@
+        token_ends=np.array([-1,  4, 8, 12, 14, 15, 17, -1]), 
+        token_starts=np.array([-1,  0,  5, 8, 13, 14, 15, -1]), 
+        useful_start=0, 
+        useful_end=512
+    )
+    label_arr = encoder.transform(out, label)
+    print(label_arr)
+    print(encoder.classes_)
+    input()
+    assert label_arr == [[0, 0, 1, 1, 0, 0, 0, 0], [0, 0, 4, 0, 5, 6, 0, 0]]
+
+def test_joint_BROS_sequence_label():
+    encoder = JointBROSEncoder(pad_token="<PAD>", bio_tagging=True)
+    labels = [
+        {'start': 0, 'end': 4, 'label': 'z', 'text': 'five'},
+        {'start': 5, 'end': 12, 'label': 'z', 'text': 'percent'},
+        {'start': 13, 'end': 17, 'label': 'z', 'text': '(5%)'},
+    ]
+    groups = [
+        {'tokens': [
+            {'start': 5, 'end': 8, 'text': 'per'},
+            {'start': 13, 'end': 17, 'text': '(5%)'},
+        ], 'label': None},
+        {'tokens': [
+            {'start': 8, 'end': 12, 'text': 'cent'},
+        ], 'label': None}
+    ]
+    label = (labels, groups)
+    encoder.fit([label])
+    print(encoder.classes_)
+    assert len(encoder.ner_classes_) == 3
+    assert len(encoder.group_classes_) == 2
+    out = EncodedOutput(
+        token_ids=np.array([   0, 9583,  139,   40,  249, 8875,    2]), 
+        tokens=np.array(['0', 'five', ' per', 'cent', ' (', '5', '%)', '2'], dtype='<U21'), 
+        token_ends=np.array([-1,  4, 8, 12, 14, 15, 17, -1]), 
+        token_starts=np.array([-1,  0,  5, 8, 13, 14, 15, -1]), 
+        useful_start=0, 
+        useful_end=512
+    )
+    label_arr = encoder.transform(out, label)
+    print(label_arr)
+    print(encoder.ner_classes_)
+    print(encoder.group_classes_)
+    input()
+    assert label_arr == [
+        [0, 1, 1, 2, 1, 2, 2, 0],
+        [0, 0, 1, 1, 0, 0, 0, 0],
+        [0, 0, 4, 0, 5, 6, 0, 0]
+    ]
+
+def test_token_relation_sequence_label():
+    encoder = TokenRelationEncoder(pad_token="<PAD>")
+    labels = [
+        {'start': 0, 'end': 4, 'label': 'z', 'text': 'five'},
+        {'start': 13, 'end': 17, 'label': 'z', 'text': '(5%)'},
+    ]
+    groups = [
+        {'tokens': [
+            {'start': 0, 'end': 4, 'text': 'five'},
+            {'start': 15, 'end': 17, 'text': '%)'},
+        ], 'label': None},
+        {'tokens': [
+            {'start': 13, 'end': 15, 'text': '(5'},
+        ], 'label': None}
+    ]
+    label = (labels, groups)
+    encoder.fit([label])
+    out = EncodedOutput(
+        token_ids=np.array([   0, 9583,  139,   40,  249, 8875,    2]), 
+        tokens=np.array(['0', 'five', ' per', 'cent', ' (', '5', '%)', '2'], dtype='<U21'), 
+        token_ends=np.array([-1,  4, 8, 12, 14, 15, 17, -1]), 
+        token_starts=np.array([-1,  0,  5, 8, 13, 14, 15, -1]), 
+        useful_start=0, 
+        useful_end=512
+    )
+    label_arr = encoder.transform(out, label)
+    print(label_arr)
+    input()
+    assert label_arr == [
+        [[0, 0, 0, 0, 0, 0, 0, 0],
```

Comment:
Having trouble deciphering what intended behavior is here. Would you mind explaining what the arrays below represent?

Reply:
Based on my understanding of the code, if arr[i][j] = 1 then the token at index i is part of the same label as j. Is this intended to be part of the same label class, or part of the same label span?